src/battle-dex-search.ts

```diff
@@ -711,6 +715,9 @@ abstract class BattleTypedSearch<T extends SearchType> {
 		while (learnsetid) {
 			let learnset = BattleTeambuilderTable.learnsets[learnsetid];
 			if (learnset && (moveid in learnset) && learnset[moveid].includes(genChar)) {
+				if (this.dex.gen >= 8 && this.dex.getMove(moveid).isNonstandard === 'Past' && this.formatType !== 'natdex') {
+					return false;
+				}
```

Comment:
Why is a nonstandard move in the Pokémon's learnset? That seems like a bug that should be fixed in the learnset rather than here.

Reply:
Oh! Nevermind, I figured it out.

Reply:
There's no reason this should be this deep into a `while` loop. Put it at the top of `canLearn`, don't repeat it for every prevo and alternate forme.